design-documents/testing-strategy.md

```diff
@@ -0,0 +1,39 @@
+## Updated Magento Commerce Testing Strategy
+
+### Terms
+
+<!-- Describe any new terms used in the document -->
+
+### Overview
+
+<!-- Describe what we are trying to solve and a few sentence summary of design approach. -->
+
+Despite running 20,000 - 30,000 tests per pull request, we have little to no confidence in the stability of the product. This means that we still need a manual regression process that lasts weeks in the best of cases. 
+
+A sizable portion of this can be attributed to our current high-level testing mentality that prioritizes system-level black box testing over other types of tests. While these types of tests are necessary and good to have, they only provide implicit coverage to the code paths of the concrete implementations that happen to be loaded in the high-level black box tests (they are also expensive/difficult to maintain). We still need explicit coverage for the concrete implementations that extension developers have hard dependencies on. This is true for the simple code paths as well as variations both positive and negative flows.
+
+### Design
+
+#### New Testing Strategy:
```

Comment:
I would recommend to run this through developers to see if it's clear. I personally think that it would be good to include some instructions and examples, it will make easier to apply the strategy.

Reply:
Yes, I agree. This proposal is not intended to be a final document. Before this PR was created the discussions surrounding this new strategy have been abstract, high-level, and theoretical. This PR was the first time it has been written down in some explicit form. There will be further PR's to come in the future with more details based on these discussions.